src/dictionary/manager.ts

```diff
@@ -20,27 +20,34 @@ const L = loggerFor(__filename);
 
 let manager: SchemaManager;
 
+type SchemaWithFields = {
+  name: string;
+  fields: string[];
+};
+
 class SchemaManager {
-  private currentSchemaDictionary: SchemasDictionary = {
-    schemas: [],
-    name: '',
-    version: '',
-  };
+  constructor(private schemaServiceUrl: string, private dictionaryName: string) {}
 
-  constructor(private schemaServiceUrl: string) {}
+  getCurrent = async (projection?: SchemasDictionaryProjection): Promise<SchemasDictionary> => {
+    const dictionary = await schemaRepo.get(this.dictionaryName, projection);
```

Comment:
instead of always fetching the full schema whenever this method is called, as I mentioned in the ticket description you can only check the version to find if a newer version was loaded, and only then do a full load if needed. the schema document can be big, did that not work or what is your opinion?

Reply:
hmm, agreed. I didn't see that in the issue and I don't see any issues with doing it that way. Will update :+1: